### PR TITLE
Remove support for using the `idea` task on the Gradle project

### PR DESCRIPTION
**Pull request by jjohannes**
The supported method to import Gradle into IntelliJ is now to use the native import of IntelliJ.

This PR...
- Removes the `apply: idea` for all (subp)rojects. The idea plugin is still configured when IntelliJ applies it during a Sync.  
- Updates the instructions for contributors
- Removes workarounds which were required to make the old import work

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -71,13 +71,7 @@ subprojects {
 
         apply(from = "../../../gradle/shared-with-buildSrc/code-quality-configuration.gradle.kts")
     }
-
-    apply(plugin = "idea")
     apply(plugin = "eclipse")
-
-    configure<IdeaModel> {
-        module.name = "buildSrc-${this@subprojects.name}"
-    }
```
Comment (jjohannes) [inline buildSrc/build.gradle.kts]: This in no longer needed. IntelliJ automatically names the modules `buildSrc.<project-name>`.

File: buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt [hunk 2]
```diff
@@ -162,68 +156,71 @@ open class IdePlugin : Plugin<Project> {
...
+            apply(plugin = "org.jetbrains.gradle.plugin.idea-ext")
+            with(model) {
+                module {
+                    excludeDirs = excludeDirs + rootExcludeDirs
                 }
-                workspace {
-                    iws {
+
+                project {
+                    jdkName = "9.0"
```
Comment (wolfs) [inline buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt]: Should be `11.0` I guess.

File: buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt [hunk 3]
```diff
@@ -162,68 +156,71 @@ open class IdePlugin : Plugin<Project> {
...
+                    wildcards.add("?*.gradle")
+                    vcs = "Git"
+                    ipr {
                         withXml {
```
Comment (wolfs) [inline buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt]: I guess we should remove the whole `withXml` call here.

**Comment by jjohannes** [comment]
@wolfs I added 4 more commits to this PR and tested it thoroughly. Everything seems to work well if you use an older wrapper (e.g. `5.2-20181221000038+0000`). With the current wrapper I hit #8244.

All outdated configuration code should now be removed from `IdePlugin.kt`.

**Pull request merged**